crucible/src/Lang/Crucible/CFG/Generator.hs

```diff
@@ -360,6 +365,20 @@ mkAtom :: (Monad m, IsSyntaxExtension ext) => Expr ext s tp -> Generator ext s t
 mkAtom (AtomExpr a)   = return a
 mkAtom (App a)        = freshAtom . EvalApp =<< traverseFC mkAtom a
 
+-- | Create a new reference bound to the value of the expression
+newReference :: (Monad m, IsSyntaxExtension ext) => Expr ext s tp -> Generator ext s t ret m (Expr ext s (ReferenceType tp))
+newReference e = do a <- mkAtom e
+                    AtomExpr <$> freshAtom (NewRef a)
+
+readReference :: (Monad m, IsSyntaxExtension ext) => Expr ext s (ReferenceType tp) -> Generator ext s t ret m (Expr ext s tp)
+readReference ref = do a <- mkAtom ref
+                       AtomExpr <$> freshAtom (ReadRef a)
+
+writeReference :: (Monad m, IsSyntaxExtension ext) => Expr ext s (ReferenceType tp) -> Expr ext s tp -> Generator ext s t ret m ()
+writeReference ref val = do aref <- mkAtom ref
+                            aval <- mkAtom val
+                            undefined -- Generator $ addStmt $ WriteRef aref aval
```

Comment:
I'm a bit worried about leaving this undefined if we are commiting it.  I don't think it is used anywhere at the moment, so we could just delete it.

Reply:
Oh, and it's one of three functions added that are all unnecessary. This one is the same as `writeRef`. I removed them all.